### ISSUE TITLE
Training Script

### ISSUE THREAD

**Issue by bsteiner-dandy** [xadrianzetx/coral-deeplab#20]
Hi @xadrianzetx, thanks again for this repository and supporting it. I am trying to create a custom model using the code in the Jupyter notebook: https://colab.research.google.com/drive/1eJFOg_1o5D-osHZseNlpE-WGR-p8jDqa?usp=sharing

It simply creates a CoralDeepLabV3 model and compiles it for the Edge TPU. However, when running it with the inference script the result is not as expected:

![segmentation_result](https://user-images.githubusercontent.com/95450720/186020796-c2e7a136-f3ee-47a5-a4db-3ce2ee4693bc.jpg)

Any idea what I could be doing wrong? Is it required to use an older version of the edgetpu-compiler if the newest version has no errors?

Pasting the code here as well:

`pip install git+https://github.com/xadrianzetx/coral-deeplab.git

! curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add -
! echo "deb https://packages.cloud.google.com/apt coral-edgetpu-stable main" | sudo tee /etc/apt/sources.list.d/coral-edgetpu.list
! sudo apt-get update
! sudo apt-get install edgetpu-compiler	
! edgetpu_compiler -v

import os
import re
import sys
import uuid
import unittest
import subprocess

import numpy as np
import tensorflow as tf

import coral_deeplab as cdl

def fake_dataset_generator(shape, n_iter):
    def dataset():
        for _ in range(n_iter):
            data = np.random.randn(*shape)
            data *= (1 / 255)
            batch = np.expand_dims(data, axis=0)
            yield [batch.astype(np.float32)]
    return dataset

def quantize_and_compile(model, dataset):

    # setup tflite converter
    converter = tf.lite.TFLiteConverter.from_keras_model(model)
    converter.representative_dataset = dataset
    converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS_INT8]
    converter.inference_input_type = tf.uint8
    converter.inference_output_type = tf.uint8
    converter.optimizations = [tf.lite.Optimize.DEFAULT]

    # quantize
    quantized = converter.convert()
    model_name = f'{uuid.uuid4()}.tflite'
    model_path = os.path.join(os.getcwd(), model_name)
    open(model_path, 'wb').write(quantized)

    # compile
    cmd = ['edgetpu_compiler', '-a', '-s', model_path]
    p = subprocess.Popen(cmd, stdout=subprocess.PIPE)
    p.wait()
    stdout = p.stdout.read()

    return stdout.decode()

model = cdl.applications.CoralDeepLabV3(weights='pascal_voc')
datagen = fake_dataset_generator((513, 513, 3), 10)
stdout = quantize_and_compile(model, datagen)
print(stdout)

! cp /content/10582856-eda3-4088-bf25-1e6a7e5e75de_edgetpu.tflite /content/drive/MyDrive/`

**Comment by xadrianzetx** [comment]
The problem seems to be in dataset generator you're using. It's just a dummy used in CI, where we do not care about model performance. Instead of using `np.random.randn`, how about feeding model actual samples from e.g. pascal voc test set?

> Is it required to use an older version of the edgetpu-compiler if the newest version has no errors?

You can probably use any compiler version as long as it maps model fully to TPU. I haven't been checking for a while, but if I remember correctly, at some point google removed `-a` flag allowing for subgraphs, hence compiler in this repo is pinned to last known working version.
